pinot-common/src/main/java/org/apache/pinot/common/utils/SegmentName.java

```diff
@@ -18,80 +18,16 @@
  */
 package org.apache.pinot.common.utils;
 
-public abstract class SegmentName {
-  public static final String SEPARATOR = "__";
-  public static final String REALTIME_SUFFIX = "_REALTIME";
-  public static final int REALTIME_SUFFIX_LENGTH = REALTIME_SUFFIX.length();
-
-  public enum RealtimeSegmentType {
-    UNSUPPORTED, HLC_LONG, HLC_SHORT, LLC,
-  }
-
-  public static RealtimeSegmentType getSegmentType(String segmentName) {
-    if (isHighLevelConsumerSegmentName(segmentName)) {
-      HLCSegmentName segName = new HLCSegmentName(segmentName);
-
-      if (segName.isOldStyleNaming()) {
-        return RealtimeSegmentType.HLC_LONG;
-      } else {
-        return RealtimeSegmentType.HLC_SHORT;
-      }
-    }
-
-    if (isLowLevelConsumerSegmentName(segmentName)) {
-      return RealtimeSegmentType.LLC;
-    }
-
-    return RealtimeSegmentType.UNSUPPORTED;
-  }
-
-  protected boolean isValidComponentName(String string) {
-    return !string.contains(SEPARATOR);
-  }
-
-  public abstract String getTableName();
-
-  public abstract String getSequenceNumberStr();
-
-  public abstract int getSequenceNumber();
-
-  public abstract String getSegmentName();
-
-  public abstract RealtimeSegmentType getSegmentType();
-
-  public String getGroupId() {
-    throw new RuntimeException("No groupId in " + getSegmentName());
-  }
-
-  public int getPartitionGroupId() {
-    throw new RuntimeException("No partitionGroupId in " + getSegmentName());
-  }
-
-  public String getPartitionRange() {
-    throw new RuntimeException("No partitionRange in " + getSegmentName());
-  }
-
-  public static boolean isHighLevelConsumerSegmentName(String segmentName) {
-    int numSeparators = getNumSeparators(segmentName);
-    return numSeparators == 2 || numSeparators == 4;
+@Deprecated
+public class SegmentName {
```

Comment:
this class can be removed?

Reply:
Kept `SegmentName` and `PartitionLevelStreamConfig` for compatibility. Will remove in next release